src/main/kotlin/com/atlassian/performance/tools/virtualusers/HttpClientWebDriver.kt

```diff
@@ -0,0 +1,334 @@
+package com.atlassian.performance.tools.virtualusers
+
+import com.google.common.util.concurrent.SettableFuture
+import net.jcip.annotations.GuardedBy
+import org.apache.http.auth.AuthScope
+import org.apache.http.auth.Credentials
+import org.apache.http.auth.UsernamePasswordCredentials
+import org.apache.http.client.CredentialsProvider
+import org.apache.http.impl.client.BasicCredentialsProvider
+import org.apache.http.impl.client.CloseableHttpClient
+import org.apache.http.impl.client.HttpClientBuilder
+import org.openqa.selenium.*
+import org.openqa.selenium.interactions.Keyboard
+import org.openqa.selenium.interactions.Mouse
+import org.openqa.selenium.remote.*
+import org.openqa.selenium.remote.internal.JsonToWebElementConverter
+import java.util.concurrent.Future
+import java.util.logging.Level
+
+internal class HttpClientWebDriver : RemoteWebDriver() {
```

Comment:
This file could be deleted when we decide to expose only a new implementation of Scenario (which uses HttpClient internally).